setup.sh

```diff
@@ -2,8 +2,26 @@
 
 CURRENT_PATH=$(pwd)
 ALIAS_COMMAND="alias git-pr-local='sh $CURRENT_PATH/git-pr-local.sh'"
+ALIAS_COMMAND_REGEX="alias git-pr-local.*"
 
-echo "\n$ALIAS_COMMAND" >> ~/.bashrc
-echo "\n$ALIAS_COMMAND" >> ~/.bash_profile
-echo "\n$ALIAS_COMMAND" >> ~/.profile
-echo "\n$ALIAS_COMMAND" >> ~/.zshrc
+declare -a SUPPORTED_TERMINAL_FILES=(".bashrc" ".bash_profile" ".profile" ".zshrc")
+
+# Run the command in current context
+eval $ALIAS_COMMAND
+
+for file in "${SUPPORTED_TERMINAL_FILES[@]}"
+do
+	# Only try adding alias to profiles when the file is readable && writable
+	if [ -w ~/"$file" ] && [ -r ~/"$file" ]
+	then
+		# Remove older added aliases before adding the new ones
+		sed -i "s/$ALIAS_COMMAND_REGEX//g" ~/"$file" 2>/dev/null
+		# For MacOS, `sed -i` expects '' argument, otherwise fails with weird errors
+		sed -i '' "s/$ALIAS_COMMAND_REGEX//g" ~/"$file" 2> /dev/null
+
+		# Add the alias
+		echo "\n$ALIAS_COMMAND" >> ~/"$file"
```

Comment:
`echo` by default adds new line at end. Let's not add new-line from outside.
```suggestion
		echo "\n$ALIAS_COMMAND" >> ~/"$file"
```